js/factory/settingsfactory.js

```diff
@@ -0,0 +1,18 @@
+app.factory('SettingsFactory', function () {
+	
+	// Using a factory as a global 'holder' for application settings applied by the user
+	// Store booleans as private vars and expose functions for getting or setting them.
+	
+	var diffMode = false;
+
+	return {
+		getMode: function () {
+			return diffMode;
+		},
+		switchMode: function () {
+			diffMode = diffMode === true ? false : true;
```

Comment:
I presume your'e referring to func 'switchMode'? I've shortedned it further actually, I'm going to just have it 'return diffMode = !diffMode;'